src/Components/v2/ArtworkFilter/ArtworkFilterContext.tsx

```diff
@@ -240,55 +243,58 @@ export const ArtworkFilterContextProvider: React.FC<
   )
 }
 
-const artworkFilterReducer = (state, action) => {
+const artworkFilterReducer = (
+  state: ArtworkFiltersState,
+  action: {
+    type: "SET" | "UNSET" | "RESET"
+    payload: { name: keyof ArtworkFilters; value?: any }
+  }
+): ArtworkFiltersState => {
   switch (action.type) {
     /**
      * Setting  and updating filters
      */
     case "SET": {
       const { name, value } = action.payload
 
-      let filterState = {}
+      let filterState: ArtworkFilters = {}
 
-      if (name === "attribution_class") {
+      if (name === "majorPeriods") {
         filterState = {
-          attribution_class: state.filters.attribution_class.concat(value),
-        }
-      }
-      if (name === "major_periods") {
-        filterState = {
-          major_periods: value ? [value] : [],
+          majorPeriods: value ? [value] : [],
         }
       }
       if (name === "page") {
         filterState[name] = Number(value)
       }
 
       // String filter types
-      ;[
+      const stringFilterTypes: Array<keyof ArtworkFilters> = [
```

Comment:
@alloy - thanks for updating with more type coverage 👍

Reply:
My pleasure 👍